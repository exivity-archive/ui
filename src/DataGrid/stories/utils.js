/**
 * Generate a unique random 8 character id (not cryptographically secure).
 *
 * @returns {string}
 */
const randomIdMemory = []
export const randomId = () => {
  let id

  if (process.env.NODE_ENV !== 'test') {
    do {
      id = Math.random().toString(36).substring(2, 10)
    } while (randomIdMemory.indexOf(id) > -1)
  } else {
    id = randomIdMemory.length
  }

  randomIdMemory.push(id)

  return id
}

/* eslint-disable-next-line max-len */
const RANDOM_WORDS = ['abstrusity', 'advertisable', 'bellwood', 'benzole', 'boreum', 'brenda', 'cassiopeian', 'chansonnier', 'cleric', 'conclusional', 'conventicle', 'copalm', 'cornopion', 'crossbar', 'disputative', 'djilas', 'ebracteate', 'ephemerally', 'epidemical', 'evasive', 'eyeglasses', 'farragut', 'fenny', 'ferryman', 'fluently', 'foreigner', 'genseng', 'glaiket', 'haunch', 'histogeny', 'illocution', 'imprescriptible', 'inapproachable', 'incisory', 'intrusiveness', 'isoceraunic', 'japygid', 'juiciest', 'jump', 'kananga', 'leavening', 'legerdemain', 'licence', 'licia', 'luanda', 'malaga', 'mathewson', 'nonhumus', 'nonsailor', 'nummary', 'nyregyhza', 'onanist', 'opis', 'orphrey', 'paganising', 'pebbling', 'penchi', 'photopia', 'pinocle', 'principally', 'prosector.', 'radiosensitive', 'redbrick', 'reexposure', 'revived', 'subexternal', 'sukarnapura', 'supersphenoid', 'tabularizing', 'territorialism', 'tester', 'thalassography', 'tuberculise', 'uncranked', 'undersawyer', 'unimpartible', 'unsubdivided', 'untwining', 'unwaived', 'webfoot', 'wedeling', 'wellingborough', 'whiffet', 'whipstall', 'wot', 'yonkersite', 'zonary']

function createRandomizedItems (depth, flat = true, parent) {
  let items = []
  let randomName
  let randomNumber
  let numChildren
  let randomValue

  if (process.env.NODE_ENV !== 'test') {
    randomName = RANDOM_WORDS[Math.floor(Math.random() * RANDOM_WORDS.length)]
    randomNumber = Math.floor(Math.random() * 100)
    numChildren = (depth < 3 && flat === false) ? Math.floor(Math.random() * 5) : 0
    randomValue = parseInt(Math.random() * 10, 10)
  } else {
    randomName = 'Test name'
    randomNumber = 50
    numChildren = (depth < 3 && flat === false) ? 3 : 0
    randomValue = 5
  }

  const random = randomId()

  const item = {
    id: random,
    parent: parent || null,
    name: randomName,
    number: randomNumber,
    group: 'Group ' + randomName.substr(0, 1),
    value: randomValue
  }

  items.push(item)

  for (let i = 0; i < numChildren; i++) {
    items = items.concat(createRandomizedItems(depth + 1, flat, item.key))
  }

  return items
}

export function createRandomizedData (items = 100, flat = true) {
  let data = []

  for (let i = 0; i < items; i++) {
    data = data.concat(createRandomizedItems(0, flat))
  }

  return data
}
