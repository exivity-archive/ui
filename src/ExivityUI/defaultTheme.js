const theme = {
  colours: {
    primary: 'mediumseagreen'
  }
}

export default theme
