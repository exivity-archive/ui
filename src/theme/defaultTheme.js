import * as constants from './constants'

const theme = {
  size: constants.SIZE_DEFAULT,
  spacing: constants.SPACING_DEFAULT,
  lineHeight: 1.5,
  fonts: {
    url: 'https://fonts.googleapis.com/css?family=Fira+Mono|Fira+Sans+Condensed:500|Fira+Sans:400,600',
    base: {
      family: '\'Fira Sans\', sans-serif',
      weight: 400,
      weightBold: 600
    },
    interaction: {
      family: '\'Fira Sans Condensed\', sans-serif',
      weight: 500
    },
    code: {
      family: '\'Fira Mono\', monospace',
      weight: 400
    }
  },
  colours: {
    lightest: '#ffffff',
    light: '#f4f4f4',

    grayLightest: '#eeeeee',
    grayLighter: '#e2e2e2',
    grayLight: '#cccccc',
    gray: '#aaaaaa',
    grayDark: '#777777',
    grayDarker: '#444444',
    grayDarkest: '#2e2e2e',

    dark: '#222222',
    darkest: '#000000',

    primaryLightest: '#e6f9ff',
    primaryLighter: '#ccf4ff',
    primaryLight: '#5cccea',
    primary: '#00a8d8',
    primaryDark: '#438dc7',
    primaryDarker: '#2e5f85',
    primaryDarkest: '#192e43',

    danger: '#c33c32',
    warning: '#ffca71',
    success: '#238e47',

    mark: '#fdf6e3'
  },
  border: {
    radius: '3px'
  }
}

export default theme
