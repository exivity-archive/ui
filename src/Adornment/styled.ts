import styled, { css } from 'styled-components'

import { SizesProps, matchThemeProp } from '../utils/styled'
import { Position } from './Adornment'
import { Block } from '../Block'

export interface StyledAdornmentProps {
  position: Position
  inset: number | string
}

export const StyledAdornment = styled.span <StyledAdornmentProps>`
  display: flex;
  align-items: center;
  position: absolute;
  height: 100%;
  top: 0;
  font-size: ${
  matchThemeProp(
    theme => theme.global.sizes,
    { modifier: (em: number) => em * 20 }
  )}px;

  ${props => props.position === Position.LEFT
    ? css`left: ${props.inset}px;`
    : css`right: ${props.inset}px;`
  }
`

export const AdornmentWrapper = styled(Block)`
  position: relative;
  display: inline-block;
`
