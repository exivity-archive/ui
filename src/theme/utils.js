import { palette, withProp } from 'styled-tools'
import color from 'color'

import theme from '.'

export const textColorWithProps = withProp(
  ['opaque', 'palette', 'tone'],
  (opaque, paletteProp, tone = 0) =>
    palette(opaque
      ? `${paletteProp}Text`
      : paletteProp, tone, 'inherit')
)

export const bgColorWithProps = withProp(
  ['opaque', 'palette', 'tone'],
  (opaque, paletteProp, tone = 0) => {
    if (!opaque) {
      return 'unset'
    }
    return palette(paletteProp, tone, 'unset')
  }
)

const hexToString = (hex) => {
  try {
    return color(hex).rgb().array().join(', ')
  } catch (err) {
    return '0, 0, 0'
  }
}

export const toCssRgbComponent = (resolver) => {
  if (typeof resolver === 'function') {
    return (props) => {
      let hex = theme.palette.primary[0]
      try {
        // @todo why the double props call?
        // console.log('props', props)
        // console.log('resolver', resolver)
        // console.log('resolver(props)', resolver(props))
        // console.log('resolver(props)(props)', resolver(props)(props))
        hex = resolver(props)(props)
      } catch (err) {
        // unset
      }
      return hexToString(hex)
    }
  } else {
    return hexToString(resolver)
  }
}
