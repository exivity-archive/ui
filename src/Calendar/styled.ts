import styled, { css } from 'styled-components'
import { fromTheme, globalFont } from '../utils/styled'

export const StyledWeekDays = styled.ul`
  ${globalFont}
  opacity: 0.6;
  margin: 0;
  padding: 10px 0;

  li {
    display: inline-block;
    width: 14.28%;
    text-align: center;
  }
`

export const StyledDays = styled.ul`
  ${globalFont}

  padding: 10px 0;
  margin: 0;

  li {
    list-style-type: none;
    display: inline-block;
    width: calc(14.28% - 5px);
    text-align: center;
    margin-right: 5px;
    margin-bottom: 5px;
  }
`

export const StyledDay = styled.div<{ active: boolean }>`
  height: 100%;
  border-radius: 180px;
  padding: 5px;
  cursor: pointer;
  user-select: none;

  &:hover {
    color: white;
    background-color: ${fromTheme(theme => theme.global.purposes.primary)};
  }

  ${({ active }) => active && css`
    color: white;
    background-color: ${fromTheme(theme => theme.global.purposes.primary)};
  `}
`
