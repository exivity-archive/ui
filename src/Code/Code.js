import PropTypes from 'prop-types'
import styled from 'styled-components'
import defaultTheme from '../theme/defaultTheme'

const Code = styled.code`
  font-family: ${props => props.theme.fonts.code.family};
  font-weight: ${props => props.theme.fonts.code.weight};
  
  background-color: ${props => props.theme.colours.mark};
  padding: 3px;
  border-radius: ${props => props.theme.border.radius};
`

Code.propTypes = {
  children: PropTypes.node.isRequired,
  theme: PropTypes.object
}

Code.defaultProps = {
  theme: defaultTheme
}

export default Code
