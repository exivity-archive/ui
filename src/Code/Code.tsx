import React, { ReactNode, useEffect, useState } from 'react'
// @ts-ignore
import SyntaxHighlighter from 'react-syntax-highlighter/dist/esm/prism-light'
// @ts-ignore
import solarizedlight from 'react-syntax-highlighter/dist/esm/styles/prism/solarizedlight'
import styled, { css } from 'styled-components'
import { BlockProps, fromTheme } from '../utils/styled'
import { preciseEm } from '../utils/styled/isolated'

interface CodeProps extends SyntaxHighlighterProps {
  language?: string
  block?: boolean
  showLineNumbers?: boolean
  children?: ReactNode
}

const useDynamicLanguage = (language: string) => {
  // Set initial state to a boolean indicating whether the language is text
  // (in which case we can skip the dynamic import effect)
  const [current, setCurrent] = useState('text')

  useEffect(() => {
    if (language !== 'text') {
      import(`react-syntax-highlighter/dist/esm/languages/prism/${language}`)
        .then(definition => {
          SyntaxHighlighter.registerLanguage(language, definition.default)
          setCurrent(language)
        })
        .catch(() => {
          console.error(`Language '${language}' could not be found, falling back to plaintext`)
          setCurrent('text')
        })
    }
  }, [language])

  return current
}

const StyledSyntaxHighlighter = styled(SyntaxHighlighter)`
  // Override globalBlockSpacing because we need !important
  margin: ${(props: BlockProps) => props.noMargin ? 0 : preciseEm(1)}rem 0 !important;

  &:first-child {
    margin-top: 0 !important;
  }

  &:last-child {
    margin-bottom: 0 !important;
  }

  & {
    background-color: ${fromTheme(theme => theme.colours.lightGray)} !important;
    &, code, pre {
      font-family: ${fromTheme(theme => theme.global.fontFamilyMonospace)} !important;
    }

    padding: ${props => props.PreTag === 'span'
      ? css`${fromTheme(theme => theme.global.spacing / 4)}`
      : css`${fromTheme(theme => theme.global.spacing)}`}em !important;
  }
`

export const Code = ({ children, block, showLineNumbers, language = 'text' }: CodeProps & BlockProps) => {
  const highlighterLanguage = useDynamicLanguage(language)

  return <StyledSyntaxHighlighter
    PreTag={block ? 'pre' : 'span'}
    style={solarizedlight}
    language={highlighterLanguage}
    showLineNumbers={showLineNumbers}
  >
    {children}
  </StyledSyntaxHighlighter>
}
