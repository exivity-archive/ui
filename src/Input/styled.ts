import styled, { css } from 'styled-components'
import { textAlign, TextAlignProps } from 'styled-system'

import { globalFont, matchThemeProp, fromTheme, hexToString, PurposesProps, SizesProps, StyledProps } from '../utils/styled'
import { Icon } from '../Icon'
import { Block } from '../Block'
import { animated } from 'react-spring'

interface ContainerProps {
  inline?: boolean
}

export const StyledContainer = styled.div<ContainerProps>`
  position: relative;

  ${props => (props.inline) && css`
    display: inline;
  `}

  ${Icon} {
    pointer-events: none;
  }
`

export interface StyledInputProps extends PurposesProps, SizesProps, StyledProps, TextAlignProps {
  // Variants
  outlined?: boolean
  flat?: boolean

  // Layout
  inline?: boolean
}

export const inputStyles = css<StyledInputProps>`
  ${globalFont};
  ${textAlign};

  font-size: ${matchThemeProp(theme => theme.global.sizes)}rem;

  box-sizing: border-box;
  padding: calc(0.5em - ${fromTheme(theme => theme.global.borderWidth)}px) 0.5em; // subtract border to get a height of exactly 2.5em for single line items

  border-radius: ${fromTheme(theme => theme.global.borderRadius)}px;
  outline: 0;
  border: 0;

  --focus-color: ${matchThemeProp(theme => theme.global.purposes, { modifier: hexToString })};

  ${props => (!props.inline) && css`
    display: block;
    width: 100%;
  `}

  ${props => (!props.outlined && !props.flat) && css`
    border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${fromTheme(theme => theme.colors.lightGray)};
    background-color: ${fromTheme(theme => theme.colors.lightGray)};

    &:hover {
      border-bottom: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 0.5);
    }

    &:focus {
      border-bottom: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 1);
    }
  `}

  ${props => (props.outlined && !props.flat) && css`
    border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${matchThemeProp(theme => theme.global.purposes)};
    background-color: unset;

    &:hover {
      border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${fromTheme(theme => theme.colors.gray)};
    }

    &:focus {
      border: ${fromTheme(theme => theme.global.borderWidth)}px solid ${fromTheme(theme => theme.colors.dark)};
    }
  `}

  ${props => props.flat && css<StyledInputProps>`
    padding: 0;

    ${props => props.outlined && css`
      outline-offset: 9px;

      &:hover {
        outline: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 0.5);
      }

      &:focus {
        outline: ${fromTheme(theme => theme.global.borderWidth)}px solid rgba(var(--focus-color), 1);
      }
    `}
  `}

  &::placeholder {
    color: currentcolor;
    opacity: 0.5;
  }

  &[disabled] {
    cursor: not-allowed;
    box-shadow: inset 0 0 999em rgba(128, 128, 128, 0.2);
  }
`

export const StyledInput = styled.input<StyledInputProps>`
  ${inputStyles}
`

export const AnimatedStyledInput = animated(StyledInput)
