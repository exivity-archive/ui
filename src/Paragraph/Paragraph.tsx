import styled from 'styled-components'
import { defaultStyledProps, globalFont, StyledProps } from '../utils/styled'

export const Paragraph = styled.p<StyledProps>`
  ${globalFont};

  margin-top: 0;

  &:last-child {
    margin-bottom: 0;
  }
`

Paragraph.defaultProps = {
  ...defaultStyledProps
}
