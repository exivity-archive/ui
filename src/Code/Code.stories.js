import React from 'react'
import { storiesOf } from '@storybook/react'

import Code from './Code'

storiesOf('atoms|Code', module)
  .add('default', () => <Code>Component</Code>)
