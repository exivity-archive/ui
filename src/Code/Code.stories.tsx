import React from 'react'
import { storiesOf } from '@storybook/react'

import { Paragraph } from '../Paragraph'

import { Code } from '.'

const code = `import { CodeBlock } from '@exivity/ui'

<CodeBlock language='jsx'>{this}</CodeBlock>`

storiesOf('atoms|Code', module)
  .add('default', () => <Paragraph>This is some <Code>inline code</Code> right here.</Paragraph>)
  .add('block', () => <Code block language='jsx'>{code}</Code>)
  .add('non-existing', () => (
    <Code block language='exivity-js'>
    Unsupported language displayed as text
    </Code>
  ))
