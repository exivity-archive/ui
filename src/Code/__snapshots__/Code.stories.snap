// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots atoms|Code block 1`] = `
.c0 {
  padding: 12px;
}

.c1 {
  margin: 1rem 0 !important;
}

.c1:first-child {
  margin-top: 0 !important;
}

.c1:last-child {
  margin-bottom: 0 !important;
}

.c1 {
  background-color: #eee !important;
  padding: 1.25em !important;
}

.c1,
.c1 code,
.c1 pre {
  font-family: 'Fira Mono',monospace !important;
}

<div
  className="c0"
>
  <pre
    className="c1"
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "4",
        "OTabSize": "4",
        "WebkitHyphens": "none",
        "backgroundColor": "#fdf6e3",
        "borderRadius": "0.3em",
        "color": "#657b83",
        "fontFamily": "Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace",
        "hyphens": "none",
        "lineHeight": "1.5",
        "margin": ".5em 0",
        "msHyphens": "none",
        "overflow": "auto",
        "padding": "1em",
        "tabSize": "4",
        "textAlign": "left",
        "whiteSpace": "pre",
        "wordBreak": "normal",
        "wordSpacing": "normal",
        "wordWrap": "normal",
      }
    }
  >
    <code
      style={
        Object {
          "MozHyphens": "none",
          "MozTabSize": "4",
          "OTabSize": "4",
          "WebkitHyphens": "none",
          "color": "#657b83",
          "fontFamily": "Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace",
          "hyphens": "none",
          "lineHeight": "1.5",
          "msHyphens": "none",
          "tabSize": "4",
          "textAlign": "left",
          "whiteSpace": "pre",
          "wordBreak": "normal",
          "wordSpacing": "normal",
          "wordWrap": "normal",
        }
      }
    >
      import { CodeBlock } from '@exivity/ui'

&lt;CodeBlock language='jsx'&gt;{this}&lt;/CodeBlock&gt;
    </code>
  </pre>
</div>
`;

exports[`Storyshots atoms|Code default 1`] = `
.c0 {
  padding: 12px;
}

.c2 {
  margin: 1rem 0 !important;
}

.c2:first-child {
  margin-top: 0 !important;
}

.c2:last-child {
  margin-bottom: 0 !important;
}

.c2 {
  background-color: #eee !important;
  padding: 0.3125em !important;
}

.c2,
.c2 code,
.c2 pre {
  font-family: 'Fira Mono',monospace !important;
}

.c1 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
  margin: 1.25rem 0;
}

.c1:first-child {
  margin-top: 0;
}

.c1:last-child {
  margin-bottom: 0;
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    This is some 
    <span
      className="c2"
      style={
        Object {
          "MozHyphens": "none",
          "MozTabSize": "4",
          "OTabSize": "4",
          "WebkitHyphens": "none",
          "backgroundColor": "#fdf6e3",
          "borderRadius": "0.3em",
          "color": "#657b83",
          "fontFamily": "Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace",
          "hyphens": "none",
          "lineHeight": "1.5",
          "margin": ".5em 0",
          "msHyphens": "none",
          "overflow": "auto",
          "padding": "1em",
          "tabSize": "4",
          "textAlign": "left",
          "whiteSpace": "pre",
          "wordBreak": "normal",
          "wordSpacing": "normal",
          "wordWrap": "normal",
        }
      }
    >
      <code
        style={
          Object {
            "MozHyphens": "none",
            "MozTabSize": "4",
            "OTabSize": "4",
            "WebkitHyphens": "none",
            "color": "#657b83",
            "fontFamily": "Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace",
            "hyphens": "none",
            "lineHeight": "1.5",
            "msHyphens": "none",
            "tabSize": "4",
            "textAlign": "left",
            "whiteSpace": "pre",
            "wordBreak": "normal",
            "wordSpacing": "normal",
            "wordWrap": "normal",
          }
        }
      >
        inline code
      </code>
    </span>
     right here.
  </p>
</div>
`;
